src/hyperlight_guest_bin/src/lib.rs

```diff
@@ -139,11 +141,39 @@ pub static mut OS_PAGE_SIZE: u32 = 0;
 // to satisfy the clippy when cfg == test
 #[allow(dead_code)]
 fn panic(info: &core::panic::PanicInfo) -> ! {
-    let msg = info.to_string();
-    let c_string = alloc::ffi::CString::new(msg)
-        .unwrap_or_else(|_| alloc::ffi::CString::new("panic (invalid utf8)").unwrap());
+    _panic_handler(info)
+}
+
+#[inline(always)]
+fn _panic_handler(info: &core::panic::PanicInfo) -> ! {
```

Comment:
I wouldn't mind putting the entire body inside `panic` instead of `_panic_handler`, but maybe it's done like this on purpose?

Reply:
Did this during dev because of the annoying rust-analyzer warning about duplicate panic handlers. I can definitely move this out!